docs/decisions/0004-axios-caching-implementation.rst

```diff
@@ -0,0 +1,87 @@
+Axios Caching Layer Implementation
+=================================================
+
+Status
+------
+
+Accepted
+
+Context
+-------
+
+The titans team noticed that on a lot of pages the same API calls are repeatedly being made as the user navigates around the website. The data being returned from these requests doesn't change between page loads leading to the same endpoints reloading the same data repeatedly. We decided that we wanted to avoid repeatedly reloading the data and reuse the data that was in the response from the initial load.
+
+We think it's beneficial to cut as much overhead as possible so each page becomes interactive more quickly.
+
+We decided to add a frontend caching layer to the AxiosJwtAuthService by leveraging an already existing npm package: `axios-cache-adapter <https://www.npmjs.com/package/axios-cache-adapter>`_
+
+Having a frontend caching layer would also allow for a stale version of a page to be displayed for a short amount of time in the case of a backend outage.
+
+When Frontend Caching Should Be Used
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The frontend cache should be used to eliminate the need to reload the same data on each page load as a user navigates around. It shouldn't be used to reduce the number of times multiple components on the same page try to make the same network call. Reducing the number of times the same network call is made on a single page should be done by moving the call higher up in the hierarchy so the call only gets made once and all lower level components are able to access the response data from it.
+
+Requests that are good candidates to use frontend caching are ones that:
+ * The data being returned changes very infrequently independently from the actions of the user on the page
+
+   - For example changes made by a background job or an admin making changes
+
+ * The data being returned isn't specific to the authenticated user
+
+ * The data returned is specific to the authenticated user but front end does not make the request when there isn't an authenticated user
```

Comment:
I'm not sure I understand what this sentence means.  Do you mean that it's a request for user-specific data that _only_ takes place when the user is authenticated?  Meaning a user is making an authenticated request for data only about themself?